### PR TITLE
Skip more OS's for AOT tests

### PR DESCRIPTION
**Pull request by BrennanConroy**

### CODE REVIEW

**Comment by eerhardt** [comment]
I think @mitchdenny's question here https://github.com/dotnet/aspnetcore/pull/48022#issuecomment-1535632641 makes a lot of sense:

> @wtgodbe is there a way with the Helix attributes that you want a test to opt into a queue rather than opt out? Seems like in the long term opting out of queues is inefficient for some test cases (although opting out is good for test cases where we know we can achieve broad platform coverage).

File: src/ProjectTemplates/test/Templates.Tests/ApiTemplateTest.cs [hunk 1]
```diff
@@ -37,7 +37,7 @@ public async Task ApiTemplateCSharp()
     }
 
     [ConditionalFact]
-    [SkipOnHelix("https://github.com/dotnet/aspnetcore/issues/47478", Queues = "OSX.13.Amd64.Open;Ubuntu.2004.Amd64.Open;Windows.11.Amd64.Client.Open;")]
+    [SkipOnHelix("https://github.com/dotnet/aspnetcore/issues/47478", Queues = "All.OSX;All.Ubuntu;Windows.11.Amd64.Client.Open;Windows.Amd64.Server2022.Open;windows.11.arm64.open")]
```
Comment (eerhardt) [inline src/ProjectTemplates/test/Templates.Tests/ApiTemplateTest.cs:40]: Just to check

> Windows.11.Amd64.Client.Open;Windows.Amd64.Server2022.Open;windows.11.arm64.open

None of those Helix queues have VS installed on them, right?

Comment (eerhardt) [reply src/ProjectTemplates/test/Templates.Tests/ApiTemplateTest.cs:40]: Should we extract this string into a constant, ex. `NativeAotSupportedHelixQueues` or similar?

Comment (BrennanConroy) [reply src/ProjectTemplates/test/Templates.Tests/ApiTemplateTest.cs:40]: Amd64.Server2022 doesn't seem to
Windows.11.Arm64 was failing with: Unable to find package Microsoft.AspNetCore.App.Runtime.win-arm64 with version (= 8.0.0-ci) so unsure if it has VS or not
Windows.11.Amd64 was already skipped. Don't know its status.

Comment (BrennanConroy) [inline src/ProjectTemplates/test/Templates.Tests/ApiTemplateTest.cs:40]:
```suggestion
    [SkipOnHelix("https://github.com/dotnet/aspnetcore/issues/47478", Queues = HelixConstants.NativeAotNotSupportedHelixQueues )]
```

File: src/ProjectTemplates/test/Templates.Tests/ApiTemplateTest.cs [hunk 2]
```diff
@@ -50,7 +50,7 @@ public async Task ApiTemplateProgramMainCSharp()
...
```
Comment (BrennanConroy) [inline src/ProjectTemplates/test/Templates.Tests/ApiTemplateTest.cs:53]:
```suggestion
    [SkipOnHelix("https://github.com/dotnet/aspnetcore/issues/47478", Queues = HelixConstants.NativeAotNotSupportedHelixQueues)]
```

**Review by wtgodbe — APPROVED** [review]

File: src/ProjectTemplates/test/Templates.Tests/ApiTemplateTest.cs (hunk 1 above)
Comment (mitchdenny) [reply src/ProjectTemplates/test/Templates.Tests/ApiTemplateTest.cs]: I've just put up a PR for a new attribute to try and make this easier moving forward:

https://github.com/dotnet/aspnetcore/pull/48117

PTAL.

**Comment by mitchdenny** [comment]
> I think @mitchdenny's question here [#48022 (comment)](https://github.com/dotnet/aspnetcore/pull/48022#issuecomment-1535632641) makes a lot of sense:
> 
> > @wtgodbe is there a way with the Helix attributes that you want a test to opt into a queue rather than opt out? Seems like in the long term opting out of queues is inefficient for some test cases (although opting out is good for test cases where we know we can achieve broad platform coverage).

Check this out:

https://github.com/dotnet/aspnetcore/pull/48117

**Pull request merged**